src/main/scala/nl/vroste/zio/kinesis/client/DynamicConsumerFake.scala

```diff
@@ -0,0 +1,135 @@
+package nl.vroste.zio.kinesis.client
+
+import java.nio.ByteBuffer
+
+import nl.vroste.zio.kinesis.client.DynamicConsumer.Checkpointer
+import nl.vroste.zio.kinesis.client.serde.{ Deserializer, Serializer }
+import software.amazon.awssdk.services.kinesis.model.EncryptionType
+import software.amazon.kinesis.common.InitialPositionInStreamExtended
+import zio._
+import zio.blocking.Blocking
+import zio.clock.Clock
+import zio.stream.ZStream
+
+private[client] class DynamicConsumerFake(
+  shards: ZStream[Any, Throwable, (String, ZStream[Any, Throwable, ByteBuffer])],
+  refCheckpointedList: Ref[Seq[Any]],
```

Comment:
Why the `Any`? Can you make it an `ExtendedSequenceNumber`?

Reply:
not sure I follow `refCheckpointedList: Ref[Seq[Any]]` holds a `Seq` of `T` where `T` is the type of the Records data eg `Record[T]`

Reply:
I see. Well, `Any` is usually not so preferable because you lose type information.

In this case, would it be sufficient to just store the checkpointed sequence numbers instead of the whole record? Then you could have a `Ref[Seq[ExtendedSequenceNumber]]`.

Reply:
The reason that `Any`/`_` is used is due to the polymorphic type signature of `shardedStream` - `T` is specified at the method level and I could not get type of Record (`T2` ) to align. The only way of getting around this was to use the intermediate type `ByteBuffer`. Also a similar approach for `Record[_]` is used in `Checkpointer`. 

IMHO I think having the whole record available is more flexible as we do not have to second guess what data the client wants.

Reply:
Fair enough